src/nginx/status.cc

```diff
@@ -270,6 +276,38 @@ Status stats_json_per_process(const ngx_esp_process_stats_t &process_stats,
   return utils::ProtoToJson(status, json, utils::JsonOptions::OUTPUT_DEFAULTS);
 };
 
+ngx_int_t ngx_esp_update_rollout(std::shared_ptr<ApiManager> esp,
+                                 ngx_esp_process_stats_t *process_stat,
+                                 int index) {
+  ServiceConfigRolloutsInfo rollouts_info;
+  esp->GetServiceConfigRollouts(&rollouts_info);
+
+  ServiceConfigRolloutsProto new_rollouts;
+  new_rollouts.set_rollout_id(rollouts_info.rollout_id);
+  for (auto percentage : rollouts_info.percentages) {
+    (*new_rollouts.mutable_percentages())[percentage.first] = percentage.second;
+  }
+
+  int length = new_rollouts.ByteSize();
+  if (0 < length && length <= kMaxServiceRolloutsInfoSize) {
+    char rollouts[kMaxServiceRolloutsInfoSize];
+    new_rollouts.SerializeToArray(rollouts, length);
+
+    ServiceConfigRolloutsProto current_rollouts;
+    if (current_rollouts.ParseFromArray(
+            process_stat->esp_stats[index].rollouts,
+            process_stat->esp_stats[index].rollouts_length)) {
```

Comment:
does ParseFromArray() return true of size is 0?   

If fail, memory is corrupted, we may want to override it